cypress/e2e/add_assurance_visit.cy.js

```diff
@@ -28,12 +28,20 @@ describe("Add Assurance Visit", () => {
 
     describe("Error submitting assurance visit form", () => {
         it("shows error message when submitting invalid data", () => {
-        cy.setCookie("fail-route", "addAssuranceVisit");
-        cy.get("#add-assurance-visit-form").submit();
-        cy.get(".govuk-error-summary__body").should(
-            "contain",
-            "Enter a requested date"
+            cy.setCookie("fail-route", "addAssuranceVisit");
+            cy.get("#add-assurance-visit-form").submit();
+            cy.get(".govuk-error-summary__body").should(
+                "contain",
+                "Select an assurance type"
+            ).should(
+                "contain",
+                "Enter a requested date"
             );
+            cy.get("#f-requested-date").should("have.class", "govuk-input--error")
```

Comment:
This isn't how we should be using ids. The `#f-` ids were only there so our JS scripts had something to select but we shouldn't be adding them just to select on in Cypress tests. 

If you have to select an element by non-visible attribute, you should be using `data-cy` attributes: https://docs.cypress.io/guides/references/best-practices#Selecting-Elements

Reply:
I'm not sure I understand - I've updated these IDs to be start with #f- as this is what their error summary links link to. From what I could tell, all the error summary links linked to IDs starting with #f-, so I needed to update the elements to start with this so the links took the user to the right input element. I can edit the tests to select the element on a different attribute than it's ID, but this needs be done across all these cypress tests so may take a while